src/nbl/ui/CCursorControlWin32.cpp

```diff
@@ -0,0 +1,41 @@
+#include <Windows.h>
+
+#include "nbl/ui/CCursorControlWin32.h"
+
+
+namespace nbl::ui
+{
+	void CCursorControlWin32::setVisible(bool visible)
+	{
+		m_windowManager->setCursorVisibility(visible);
+	}
+	bool CCursorControlWin32::isVisible() const
+	{
+		CURSORINFO ci = { sizeof(CURSORINFO) };
+		GetCursorInfo(&ci);
+		return ci.flags; // returning flags cause they're equal to 0 when the cursor is hidden
+	}
+	void CCursorControlWin32::setPosition(int32_t x, int32_t y)
+	{
+		SetCursorPos(x, y);
```

Comment:
doesn't this need to go through the window manager's thread caller?

Reply:
nah, only visibility setting needs to go through that thread. The reason for that is that they are directly connected to the window (so when you set the cursor to be invisible, it's only gonna be invisible inside the window rectangle). other calls are just winapi calls that have nothing in common with the window

Reply:
okay, but can these WinAPI calls be hammered through threads and possibly concurrently?

Reply:
yeah, why not. Windows seems to only have thread issues with calls that send messages to a specific window. in that case you need to do all that in a thread window was created in (cause the message sending call and a window must share a message queue, which is one per thread)

Reply:
I mean there's an easy way to check, spawn 32 threads, and let them hammer `Set` and `Get` CursorPos, if it doesnt crash, we're good

Reply:
didi you check?

Reply:
yeah, just checked it. no crashes